libsession/src/main/java/org/session/libsession/snode/SnodeAPI.kt

```diff
@@ -247,190 +206,151 @@ object SnodeAPI {
 
     // Public API
     fun getAccountID(onsName: String): Promise<String, Exception> {
-        val deferred = deferred<String, Exception>()
-        val promise = deferred.promise
         val validationCount = 3
         val accountIDByteCount = 33
         // Hash the ONS name using BLAKE2b
         val onsName = onsName.toLowerCase(Locale.US)
         val nameAsData = onsName.toByteArray()
         val nameHash = ByteArray(GenericHash.BYTES)
         if (!sodium.cryptoGenericHash(nameHash, nameHash.size, nameAsData, nameAsData.size.toLong())) {
-            deferred.reject(Error.HashingFailed)
-            return promise
+            throw Error.HashingFailed
```

Comment:
I get that the _current_ receiving functions of `getAccountID` are in try/catch but I'm not sure which should be mixing "philosophies" here. If this is still using the ol' Promises, then it should probably be handled with that mechanism instead of throwing

Reply:
`interface Promise<out V, out E>` has many implementations. In this part I've swapped a deferred for a task.

With a Deferred you call `resolve` or `reject`

In `task(..., body: () -> V)` you return in the lambda or throw

task has narrower more clearly defined scope than deferred. Using deferred implies that we will pass it somewhere else, but this does not occur, task is clearer.

Reply:
where is task created there though?

Reply:
later, yeah, that's an issue... I could return `Promise.ofFailure` here, or would have to make this whole function a task and then bind to the next `promises.all` thing.

The former sounds better.